controllers/drcluster_controller.go

```diff
@@ -277,6 +272,44 @@ func filterDRClusterMCV(mcv *viewv1beta1.ManagedClusterView) []ctrl.Request {
 	}
 }
 
+func filterDRClusterSecret(reader client.Reader, secret *corev1.Secret) []ctrl.Request {
+	log := ctrl.Log.WithName("filterDRClusterSecret").WithName("Secret")
+
+	drcusters := &ramen.DRClusterList{}
+	if err := reader.List(context.TODO(), drcusters); err != nil {
+		log.Error(err, "Failed to get list of DRCluster resources")
+
+		return []reconcile.Request{}
+	}
+
+	requests := []reconcile.Request{}
+
+	for i := range drcusters.Items {
+		drcluster := &drcusters.Items[i]
+
+		s3ProfileName := drcluster.Spec.S3ProfileName
+
+		if s3ProfileName != NoS3StoreAvailable {
```

Comment:
It would be nicer to continue in this case - keeping the happy path convention.

    if s3ProfileName == NoS3StoreAvailable {
        continue
    }

    handle it...

Reply:
Done